app/controllers/nonsipp/membersurrenderedbenefits/SurrenderedBenefitsCYAController.scala

```diff
@@ -0,0 +1,187 @@
+            .getOrRecoverJourney
+        } yield Ok(
+          view(
+            viewModel(
+              srn,
+              memberIndex,
+              memberDetails.fullName,
+              surrenderedBenefitsAmount,
+              whenSurrenderedBenefits,
+              whySurrenderedBenefits,
+              mode
+            )
+          )
+        )
+      ).merge
+    }
+
+  def onSubmit(srn: Srn, memberIndex: Max300, mode: Mode): Action[AnyContent] =
+    identifyAndRequireData(srn).async { implicit request =>
+      psrSubmissionService.submitPsrDetails(srn).map {
+        case None =>
+          Redirect(controllers.routes.JourneyRecoveryController.onPageLoad())
+        case Some(_) =>
+          Redirect(navigator.nextPage(SurrenderedBenefitsCYAPage(srn, memberIndex), mode, request.userAnswers))
+      }
+    }
+}
+
+object SurrenderedBenefitsCYAController {
+  def viewModel(
+    srn: Srn,
+    memberIndex: Max300,
+    memberName: String,
+    surrenderedBenefitsAmount: Money,
+    whenSurrenderedBenefits: LocalDate,
+    whySurrenderedBenefits: String,
+    mode: Mode
+  ): FormPageViewModel[CheckYourAnswersViewModel] =
+    FormPageViewModel[CheckYourAnswersViewModel](
+      title = mode.fold(
+        normal = "surrenderedBenefits.cya.title",
+        check = "surrenderedBenefits.change.title"
+      ),
+      heading = mode.fold(
+        normal = "surrenderedBenefits.cya.heading",
+        check = Message("surrenderedBenefits.change.heading", memberName)
+      ),
+      description = None,
+      page = CheckYourAnswersViewModel(
+        sections = rows(
+          srn,
+          memberIndex,
+          memberName,
+          surrenderedBenefitsAmount,
+          whenSurrenderedBenefits,
+          whySurrenderedBenefits
+        )
+      ),
+      refresh = None,
+      buttonText = mode.fold(
+        normal = "site.saveAndContinue",
+        check = "site.continue"
+      ),
+      onSubmit = routes.SurrenderedBenefitsCYAController.onSubmit(srn, memberIndex, mode)
+    )
+
+  private def rows(
+    srn: Srn,
+    memberIndex: Max300,
+    memberName: String,
+    surrenderedBenefitsAmount: Money,
+    whenSurrenderedBenefits: LocalDate,
+    whySurrenderedBenefits: String
+  ): List[CheckYourAnswersSection] =
+    List(
+      CheckYourAnswersSection(
+        None,
+        List(
+          CheckYourAnswersRowViewModel(
+            Message("surrenderedBenefits.cya.section.memberName"),
+            Message(memberName)
+          ),
+          CheckYourAnswersRowViewModel(
+            Message("surrenderedBenefits.cya.section.amount", memberName),
+            Message(s"£${surrenderedBenefitsAmount.displayAs}")
+          ).withAction(
+            SummaryAction(
+              "site.change",
+              controllers.nonsipp.membersurrenderedbenefits.routes.SurrenderedBenefitsAmountController
+                .onSubmit(srn, memberIndex, CheckMode)
+                .url + "#surrenderedBenefitsAmount"
```

Comment:
This kind of markup with '#<field name>' is only needed if the page has multiple fields. In this case none of these 3 values require that.